<launch>

  <arg name="namespace" default="Bilbo" />

  <node pkg="px4_control"
    type="px4_control_node"
    name="px4_control_node"
    output="screen">

    <!-- Namespace -->
    <param name="namespace" value="$(arg namespace)"/>     <!-- degrees -->

    <!-- Joystick Attitude Reference parameters -->
    <param name="RollMax" value="30.0"/>     <!-- degrees -->
    <param name="PitchMax" value="30.0"/>    <!-- degrees -->
    <param name="YawRateMax" value="45.0"/>  <!-- deg/sec -->
    <param name="maxThrust" value="0.7"/>    <!-- From 0 to 1 -->

    <!-- Joystick Position Reference parameters -->
    <param name="xRate" value="0.5"/>       <!-- meters/sec -->
    <param name="yRate" value="0.5"/>       <!-- meters/sec -->
    <param name="zRate" value="0.5"/>       <!-- meters/sec -->
    <param name="PosRefTimeConstant" value="0.2"/> <!-- seconds -->

    <!-- Topic with odometry information for the vehicle -->
    <param name="odomTopic" value="mavros/local_position/odom"/>

    <!-- Name of topic to use when handling PVA_References-->
    <param name="pvaTopic" value="px4_control/PVA_Ref"/>

    <!-- Type of joystick being used. Admitted types are:
         joyXboxOne  /  joyXbox360  /  joyXbox360Wired -->
    <param name="joyDriver" value="joyXboxOne"/>

    <!-- Position Control parameters -->
    <param name="mass" value="1.535"/>       <!-- kg -->
    <param name="gz" value="9.81"/>          <!-- m/s^2 -->
    <param name="thrustRatio" value="1.7"/>  <!-- maxThrust/Weight -->
    <param name="kpx" value="10"/>
    <param name="kpy" value="10"/>
    <param name="kpz" value="10"/>
    <param name="kvx" value="5"/>
    <param name="kvy" value="5"/>
    <param name="kvz" value="7.5"/>
    <param name="kix" value="0"/>
    <param name="kiy" value="0"/>
    <param name="kiz" value="0"/>
    <param name="maxInteg_x" value="0"/>
    <param name="maxInteg_y" value="0"/>
    <param name="maxInteg_z" value="0"/>

    <!-- Autoland parameters -->
    <param name="land_speed" value="0.5"/>    <!-- m/s -->

    <!-- Timeouts to trigger autoland -->
    <param name="joystickTimeout" value="1.0"/>
    <param name="odometryTimeout" value="1.0"/>
  </node>
</launch>
